src/api/models/publishedFacet.js

```diff
@@ -0,0 +1,43 @@
+import chunk from 'lodash.chunk';
+
+export default (db) => {
+    const collection = db.collection('publishedFacet');
+
+    collection.insertBatch = documents => chunk(documents, 100).map(data => collection.insertMany(data));
+
+    collection.insertFacet = (field, values) =>
+        collection.insertBatch(values.map(value => ({
+            field,
+            value,
+        })));
+
+    collection.findLimitFromSkip = (limit, skip, filter) =>
+        collection
+            .find(filter)
+            .skip(skip)
+            .limit(limit)
+            .sort({ value: 1 })
```

Comment:
I'm quite sure that's not true as the query is only executed when yielding `toArray`